src/app/ui/color_bar.cpp

```diff
@@ -1182,9 +1183,16 @@ void ColorBar::updateCurrentSpritePalette(const char* operationName)
           cmd->execute(UIContext::instance());
         }
         else {
-          Tx tx(writer.context(), operationName, ModifyDocument);
-          tx(cmd);
-          tx.commit();
+          std::unique_ptr<Tx> tx;
+          try {
+            tx = std::unique_ptr<Tx>(new Tx(writer.context(), operationName, ModifyDocument));
+          }
+          catch (...) {
+            delete cmd;
```

Comment:
It's a known issue that in some places there are some leaks on some `Cmd`s before adding then to the `Tx`. We think that we need a more general solution to this problem (e.g. creating a custom `CmdPtr` which could be an alias/similar to `std::unique_ptr`).